Classes/Command/FluidTemplatePrecompileCommand.php

```diff
@@ -0,0 +1,161 @@
+<?php
+namespace NamelessCoder\CmsFluidPrecompilerModule\Command;
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Helper\TableSeparator;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use TYPO3Fluid\Fluid\Core\Cache\FluidCacheWarmupResult;
+
+/**
+ * A console command for precompiling Fluid templates
+ */
+class FluidTemplatePrecompileCommand extends Command
+{
+    use FluidPrecompilerServiceTrait;
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setDescription('Precompile a Fluid resource')
+            ->addArgument(
+                'extension',
+                InputArgument::OPTIONAL,
+                'Extension key of which the Fluid resources should be compiled. Leave empty to compile all',
+                null
+            )
+            ->addOption(
+                'fail',
+                null,
+                InputOption::VALUE_NONE,
+                'Fail with a non-zero return code if any template could not be compiled'
+            )
+            ->addOption(
+                'show-only-failed',
+                null,
+                InputOption::VALUE_NONE,
+                'Show only failed Fluid templates which failed to compile (in conjunction with -v)'
+            )
+            ->addOption(
+                'limit',
+                'l',
+                InputOption::VALUE_REQUIRED,
+                'Limit template information shown of each extension (in conjunction with -v)',
+                5
+            )
+            ->addOption(
+                'disable-error-handler',
+                null,
+                InputOption::VALUE_NONE,
+                'Temporarily disable any registered error handler for compiling'
+            );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $outputHelper = new SymfonyStyle($input, $output);
+
+        // Get arguments and options
+        $extension = $input->getArgument('extension');
+        $fail = $input->getOption('fail');
+        $onlyFailed = $input->getOption('show-only-failed');
+        $limit = $input->getOption('limit');
+        $disableErrorHandlers = $input->getOption('disable-error-handler');
+
+        // Test exception handlers
+        if (set_error_handler(function() {}) && !$disableErrorHandlers) {
```

Comment:
This really should not be necessary - the default behavior is to catch exceptions internally and analyse them. We should not need to replace the error handler: if an exception really is raised, it is an error in the templates or an incompatibility which prevents it from being parsed without errors.

FYI: Fluid has an internal error handling device but it is *not* a PHP registered error handler. It is quite selective about which exceptions are actual errors and which are parsing errors - and only "real" errors would break in a way that triggers whichever error handler is registered. Additionally, PHP warnings must definitely not be suppressed because they indicate real errors (such as not checking input types, abusing parse-time execution, etc.)

Therefore I would rather we do not make it possible to manipulate this low level of the runtime. The right way is to show the errors and expect the user to fix them.